pkg/grafana/grafana.go

```diff
@@ -61,20 +62,22 @@ var defaultDatasources = []Datasource{
 	},
 }
 
-func UpsertOrganization(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, organization *Organization) error {
+func UpsertOrganization(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, organization *Organization) (err error) {
 	logger := log.FromContext(ctx)
+	logger.Info("upserting organization")
 
-	err := assertNameIsAvailable(ctx, grafanaAPI, organization)
-	if err != nil {
-		return errors.WithStack(err)
+	// Get the current organization stored in Grafana
+	var currentOrganization *Organization
+	if organization.ID != 0 {
+		currentOrganization, err = findOrgByID(grafanaAPI, organization.ID)
```

Comment:
Also this will not work after grafana restarted because all organizations will bé gone so this id and even the name would not bé there

Reply:
If the organization is not found (either by name of id) it is then created